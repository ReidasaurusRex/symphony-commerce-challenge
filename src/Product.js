import React, { Component } from 'react';

class Product extends Component {
  render() {
    return (
      <li className="c-product">
        <h2 className="c-product__name">{this.props.name}</h2>
        <p className="c-product__price">{this.props.price}</p>
        <img src={this.props.image} alt={this.props.name} className="c-product__image"/>
      </li>
    );
  }
}

export default Product;