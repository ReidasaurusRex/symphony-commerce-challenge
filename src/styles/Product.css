.c-product {
  margin-top: 50px;
}

.c-product__text, .c-product__image {
  display: inline-block;
  vertical-align: top;
}

.c-product__text {
  margin-left: 25px;
  margin-top: 150px;
  width: 350px;
}

@media (max-width: 755px) {
  .c-product__text {
    margin-top: 25px;
    text-align: center;
  }
}