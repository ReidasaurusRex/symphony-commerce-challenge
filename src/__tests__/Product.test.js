import React, { Component } from 'react';
import Product from "../Product";
import renderer from 'react-test-renderer';

test('Product renders name, price, and image from props', () => {
  const product = {
    name: "Product 1",
    price: 100,
    image: "image-cdn.symphonycommerce.com/images/sites/fijiwater/1424736511908_6pack_straw_image_1.400w.jpg"
  }

  const component = renderer.create(
    <Product name={product.name} price={product.price} image={product.image} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});