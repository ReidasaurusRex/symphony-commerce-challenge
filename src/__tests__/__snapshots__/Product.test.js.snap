exports[`test Product renders name, price, and image from props 1`] = `
<li
  className="c-product">
  <h2
    className="c-product__name">
    Product 1
  </h2>
  <p
    className="c-product__price">
    100
  </p>
  <img
    alt="Product 1"
    className="c-product__image"
    src="image-cdn.symphonycommerce.com/images/sites/fijiwater/1424736511908_6pack_straw_image_1.400w.jpg" />
</li>
`;
